### PR TITLE
Implement undo command

### PR DESCRIPTION
**Pull request by vadmeste**
mc undo removes the last added version of a specific object or objects under a prefix.

mc undo requires a enabled versioned bucket to work.

### CODE REVIEW

File: cmd/undo-main.go [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+USAGE:
+  {{.HelpName}} [FLAGS] SOURCE [SOURCE...]
+{{if .VisibleFlags}}
+FLAGS:
+  {{range .VisibleFlags}}{{.}}
+  {{end}}{{end}}
+
+EXAMPLES:
+  1. Undo the last 3 uploads and/or removals of a particular object
+     {{.Prompt}} {{.HelpName}} s3/backups/file.zip
```
Comment (kannappanr) [inline cmd/undo-main.go:73]:
```suggestion
     {{.Prompt}} {{.HelpName}} s3/backups/file.zip --last 3
```

**Comment by kannappanr** [comment]
```
$ mc mb play/undo
Bucket created successfully `play/undo`.

$ mc version enable play/undo
play/undo versioning is enabled

$ mc cp /etc/issue play/undo/
/etc/issue:                          26 B / 26 B ┃▓▓▓▓▓▓▓▓┃ 331 B/s 0s

$ mc cp /etc/issue play/undo/
/etc/issue:                          26 B / 26 B ┃▓▓▓▓▓▓▓▓┃ 208 B/s 0s
$ mc cp /etc/issue play/undo/
/etc/issue:                          26 B / 26 B ┃▓▓▓▓▓▓▓▓┃ 280 B/s 0s
 $ mc rm play/undo/issue 
Removing `play/undo/issue`.
 $ mc ls play/undo/issue --versions
mc: <ERROR> Unable to stat `play/undo/issue`. Object did not exist at `Tue, 15 Sep 2020 07:47:10 UTC`.

```

`mc ls --versions` has an issue in this PR, when the latest version is a delete marker

**Comment by kannappanr** [comment]
```
$ mc ls --versions play/undo 
[2020-09-15 00:53:58 PDT]     26B issue (4th, vid=0fb50a4b-6e43-44a9-b130-611f24d0e1ec)
[2020-09-15 00:53:57 PDT]     26B issue (3rd, vid=7b78b5d6-c619-4a5d-9b64-d3afc8f23214)
[2020-09-15 00:53:56 PDT]     26B issue (2nd, vid=d33601de-96de-4ada-9ec5-9541f34c11c9)
[2020-09-15 00:53:55 PDT]     26B issue (1st, vid=d4f6aacd-4811-4053-8b94-8671da5814e5)
$ mc undo play/undo 
mc: <ERROR> Unable to find any object version to undo. 
```

**Comment by vadmeste** [comment]
> `mc ls --versions` has an issue in this PR, when the latest version is a delete marker

This is a bug which is already fixed in master, I rebased this PR, the bug doesn't exist anymore

**Comment by vadmeste** [comment]
> $ mc ls --versions play/undo 
> [2020-09-15 00:53:58 PDT]     26B issue (4th, vid=0fb50a4b-6e43-44a9-b130-611f24d0e1ec)
> [2020-09-15 00:53:57 PDT]     26B issue (3rd, vid=7b78b5d6-c619-4a5d-9b64-d3afc8f23214)
> [2020-09-15 00:53:56 PDT]     26B issue (2nd, vid=d33601de-96de-4ada-9ec5-9541f34c11c9)
> [2020-09-15 00:53:55 PDT]     26B issue (1st, vid=d4f6aacd-4811-4053-8b94-8671da5814e5)
> $ mc undo play/undo 
> mc: <ERROR> Unable to find any object version to undo.

This also seems to be fixed.

**Review by poornas — CHANGES_REQUESTED** [review]
Needs docs, other than that, tested and LGTM

**Review by kannappanr — APPROVED** [review]
except for the changes suggested by Poorna , LGTM

**Comment by vadmeste** [comment]
> Needs docs, other than that, tested and LGTM

@poornas done

**Review by poornas — APPROVED** [review]
LGTM

**Pull request merged**